src/lib/design/Badge.tsx

```diff
@@ -1,36 +1,60 @@
-import styled from "styled-components";
+import styled, {css} from "styled-components";
 import {receiverTheme} from "@/lib/design/wip/receiverTheme";
 import {textXsMedium} from "@/lib/design/wip/typography";
 
 
 const Root = styled.div`
   display: flex;
 
-  width: 100%;
-`;
+  width: 100%;`;
 
 const Container = styled.div`
-  ${textXsMedium};
-  color: ${receiverTheme.colors.gray["700"]};
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  column-gap: 0.4rem;
+
   border-radius: 16px;
   background-color: ${receiverTheme.colors.gray["100"]};
   padding: 2px 8px;
   height: 1.125rem;
+  min-width: 3rem;
+`;
 
-  display: flex;
-  justify-content: center;
-  align-items: center;
+const Label = styled.div<{ color: "gray" | "purple" }>`
+  ${textXsMedium};
+  color: ${receiverTheme.colors.primary["500"]};
+
+  ${props => props.color === "gray" && css`
+    color: ${receiverTheme.colors.gray["700"]}
+  `};
+`;
+
+const Icon = styled.div<{ color: "gray" | "purple" }>`
+  width: 0.375rem;
+  height: 0.375rem;
+  border-radius: 50%;
+  background-color: ${receiverTheme.colors.primary["500"]};
+
+  ${props => props.color === "gray" && css`
+    background-color: ${receiverTheme.colors.gray["700"]}
+  `};
 `;
 
 const LoadingDiv = styled.div`
```

Comment:
Background doesn't look the same as the design. If we use color gradient, we need the add one more line to our css.
`mix-blend-mode: multiply;`